polygeist/tools/cgeist/Lib/CodeGenTypes.cc

```diff
@@ -1339,35 +1344,34 @@ mlir::Type CodeGenTypes::getMLIRType(clang::QualType QT, bool *ImplicitRef,
     }
 
     const auto *RD = RT->getAsRecordDecl();
-    if (mlirclang::isNamespaceSYCL(RD->getEnclosingNamespaceContext())) {
-      const auto TypeName = RD->getName();
-      if (TypeName == "accessor" || TypeName == "accessor_common" ||
-          TypeName == "AccessorImplDevice" || TypeName == "AccessorSubscript" ||
-          TypeName == "array" || TypeName == "AssertHappened" ||
-          TypeName == "atomic" || TypeName == "bfloat16" ||
-          TypeName == "GetOp" || TypeName == "GetScalarOp" ||
-          TypeName == "group" || TypeName == "h_item" || TypeName == "id" ||
-          TypeName == "item" || TypeName == "ItemBase" ||
-          TypeName == "kernel_handler" ||
-          TypeName == "LocalAccessorBaseDevice" ||
-          TypeName == "local_accessor_base" || TypeName == "local_accessor" ||
-          TypeName == "maximum" || TypeName == "minimum" ||
-          TypeName == "multi_ptr" || TypeName == "nd_item" ||
-          TypeName == "nd_range" || TypeName == "OwnerLessBase" ||
-          TypeName == "range" || TypeName == "stream" ||
-          TypeName == "sub_group" || TypeName == "SwizzleOp" ||
-          TypeName == "TupleCopyAssignableValueHolder" ||
-          TypeName == "TupleValueHolder" || TypeName == "vec")
-        return getSYCLType(RT, *this);
-
-      // No need special handling for types that don't have record declaration
-      // name.
-      CGEIST_WARNING({
-        if (TypeName != "")
-          llvm::WithColor::warning()
-              << "SYCL type '" << ST->getName()
-              << "' has not been converted to SYCL MLIR\n";
-      });
+    {
```

Comment:
are there any reasons to create a new scope here?

Reply:
To limit the scope of `IsNamespaceSYCL`. I can remove if you think it'd be better.

Reply:
Is fine, you can keep it, just want to learn if there are other benefits I am not aware of.

Reply:
Now that I learn about C++17's init-statement, we can use it here too 😉

Reply:
Would make sense. Done!